### PR TITLE
Run lint and tests in Buildkite CI

### PR DESCRIPTION
**Pull request by mokagio**
Moves the lint and unit tests CI steps from CircleCI to Buildkite.

It uses an experimental approach to `npm` that:

1. Runs `npm ci` instead of `npm install`, as recommended for CI jobs
2. Uses caching, but since we're using `npm ci`, which recreates `node_modules`, only caches the NPM cache.
3. Has a gate at the start of the pipeline to create the cache if necessary, so that we'll avoid race conditions where multiple jobs might attempt to upload the NPM cache simultaneously.

I'm a bit unsure about the approach to be honest. Is the cache gate necessary? What are the chances of race condition? Besides, the performance of using vs not using the cache with `npm ci` do not seem to be that different, which makes me question whether it's worth having in the first place.

Keen to hear what you think. I'd still propose to merge as is and see how it behaves. This investigation has a broader scope that just this project, as we have new Electron apps that might use this new approach. 

### Test
See green CI.

### Release
N.A.

### CODE REVIEW

File: .buildkite/commands/install_node_dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -eu
+
+# --legacy-peer-deps is necessary because of react-monaco-editor.
+# See README for more details
+install_npm_packages_clean --legacy-peer-deps
```
Comment (mokagio) [inline .buildkite/commands/install_node_dependencies.sh:5]: See https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/96

**Comment by mokagio** [comment]
Requested an @Automattic/apps-infrastructure review for feedback on the `npm` approach.

@codebykat the head is `release/2.22.0` because I'm wishfully hoping to port the rest of the setup before it's time to cut the final build. At worst, we'll ship 2.22.0 on the old infra and continue the upgrade later.

File: .xcode-version [hunk 2]
```diff
@@ -0,0 +1 @@
+15.4
```
Comment (codebykat) [inline .xcode-version:1]: This snuck in here I think? Unless xcode is somehow required for Buildkite. I'd prefer this be removed and added to `.gitignore` if you like.

**Review by codebykat — COMMENTED** [review]
Okay by me, if it works - I don't have an opinion about the npm caching. I know CircleCI is doing a bunch of stuff to sign the builds already, but it's finicky, maybe Buildkite is less finicky (however I have not worked with Buildkite at all). I'm happy for us to do whatever integrates best with the apps infra processes.

I did leave a note about the `.xcode-version` which probably shouldn't be there.

File: .xcode-version (hunk 2 above)
Comment (mokagio) [reply .xcode-version:1]: This is here intentionally. It's sort of like `.nvmrc`

It is "required" in that Buildkite reads it to know which macOS-Xcode VM to load. Other tooling might read this, although I don't think any would at the moment.

https://github.com/Automattic/simplenote-electron/pull/3206/files#diff-0259de1994ebca16ff7f60a3198680b690e517788c1b01be7c039206801f6ce5R7

**Comment by mokagio** [comment]
Superseded by #3204

**Pull request closed without merging**